halo2-wasm/src/halo2lib.rs

```diff
@@ -329,6 +330,36 @@ impl Halo2LibWasm {
         self.to_js_assigned_values(out)
     }
 
+    pub fn to_hi_lo(&mut self, a: usize) -> Vec<u32> {
+        let a = self.get_assigned_value(a);
+        let a_val = a.value();
+        let a_bytes = a_val.to_bytes();
+
+        let mut a0_bytes = [0u8; 32];
+        let mut a1_bytes = [0u8; 32];
+        a0_bytes[..16].copy_from_slice(&a_bytes[..16]);
+        a1_bytes[..16].copy_from_slice(&a_bytes[16..]);
+        let a0 = Fr::from_bytes(&a0_bytes).unwrap();
+        let a1 = Fr::from_bytes(&a1_bytes).unwrap();
+
+        let mut builder = self.builder.borrow_mut();
+        let ctx = builder.main(0);
+
+        let a0 = ctx.load_witness(a0);
+        let a1 = ctx.load_witness(a1);
+
+        self.range.range_check(ctx, a0, 128);
+        self.range.range_check(ctx, a1, 125);
+
+        let two_pow_128 = ctx.load_witness(self.gate.pow_of_two()[128]);
```

Comment:
this needs to be `load_constant`

Reply:
how you can directly put `QuantumCell::Constant(self.gate.pow_of_two()[128])` into `mul_add` below, and it saves a load.